[NgZiXin] iP

NgZiXin opened the pull request:
## Week 2 IP Progress

### Commands Available

You can call the following commands:

- **list**
- **todo** description
- **deadline** description _/by_ date
- **event** description _/from_ date _/to_ date
- **unmark** 1
- **mark** 1
- **bye**

### Code Sample

My code is written in Java. This is the main function:

```java
public static void main(String[] args) {
    (new Ollie("./data/tasks.txt")).run();
}
```

## Task List for Week 4

1. Update PR description
2. Review Peer PR
3. Learn from others
4. Add Increments as branches: `A-CheckStyle`, `Level-10`, `A-Varargs`

## Tasks to Complete

- [x] Update PR description
- [ ] Review Peer PR
- [ ] Learn from others
- [ ] Add Increments as branches: `A-CheckStyle`, `Level-10`, `A-Varargs`

### Note to myself

🚀  Don't forget to submit my progress by the end of the week!

> "Success is not the key to happiness. Happiness is the key to success. If you love what you are doing, you will be successful." — Albert Schweitzer

### Reference

For more information, check the [GitHub Markdown Guide](https://guides.github.com/features/mastering-markdown/).

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

NgZiXin committed:
Renamed Bot, Added Greet and Exit functionalities

NgZiXin committed:
Added Echo Functionality

NgZiXin committed:
Added List Functionality

NgZiXin committed:
Refactor code for modularization

NgZiXin committed:
Added Mark As Done and Unmark

NgZiXin committed:
Added Deadline, Todo, Event class

NgZiXin committed:
Fixed bug with Todo Class

NgZiXin committed:
Add insert deadline, todo, event feature

NgZiXin committed:
Fixed formatting bug in Ollie

NgZiXin committed:
Configure semi-automated I/O tests

NgZiXin committed:
Add Exception Handler

NgZiXin committed:
Add delete fucntionality

NgZiXin committed:
Add load and save functionalities

NgZiXin committed:
Merge branch 'branch-Level-7'

NgZiXin committed:
Add Storage class

NgZiXin committed:
Add Ui class skeleton

NgZiXin committed:
Refactor code to follow OOP

NgZiXin committed:
Divide classes into Package

NgZiXin committed:
Update add-gradle-support branch

NgZiXin committed:
Merge branch 'add-gradle-support' of https://github.com/NgZiXin/ip into add-gradle-support

NgZiXin committed:
Merge branch 'add-gradle-support'

NgZiXin committed:
Add J-Unit test cases

NgZiXin committed:
Add date time parsing to chatbot

NgZiXin committed:
Merge with branch-Level-8

NgZiXin committed:
Update .gitignore

NgZiXin committed:
Add JavaDocs to all classes

NgZiXin committed:
Add find task feature

NgZiXin committed:
Update code to meet coding standards

NgZiXin committed:
Merge with branch-A-CodingStandard

NgZiXin committed:
Merge with branch-Level-9

NgZiXin committed:
Merge branch 'branch-A-JavaDoc' into HEAD

NgZiXin committed:
Merge with branch-A-CodingStandard

NgZiXin committed:
Merge with branch-Level-9

NgZiXin committed:
Fix coding style

NgZiXin committed:
Fix coding style

wkxcass commented on src/main/java/ollie/Ui.java:
```diff
@@ -0,0 +1,95 @@
+    private final Scanner scanner = new Scanner(System.in);
+
+    /**
+     * Read and return the user's input.
```
Perhaps changing "Read and return..." into "Reads and returns..."? I noticed similar issues in some other places too!

wkxcass commented on src/main/java/ollie/command/FindCommand.java:
```diff
@@ -0,0 +1,20 @@
+import ollie.task.Task;
+
+public class FindCommand extends Command{
+    String findQuery;
```
Perhaps renaming `findQuery` into something more descriptive, such as `keyword`? I find the term "find" a bit ambiguous as it could be interpreted either as a noun (referring to the "find" command) or as a verb (implying the action of finding a query)! Also noticed this in a few other places~

wkxcass commented on src/main/java/ollie/command/MarkCommand.java:
```diff
@@ -0,0 +1,30 @@
+package ollie.command;
+
+import ollie.*;
```
Perhaps listing out the imported classes explicitly? I noticed similar issues in several other places too!

minhao23 replied:
try not to use the wildcard import!

wkxcass commented on src/test/java/ollie/ParserTest.java:
```diff
@@ -0,0 +1,75 @@
+public class ParserTest {
+
+    @Test
+    public void parse_NotRecognisedCommand_exceptionThrown() {
```
Perhaps changing the uppercase "N" into a lowercase "n"?

wkxcass commented on src/main/java/ollie/task/Task.java:
```diff
@@ -0,0 +1,53 @@
+        return String.format("%d | %s", isDone ? 1 : 0, this.description);
+    }
+
+    public boolean doesDescContain(String findQuery) {
```
I kind of like the naming of this `boolean` returning method!

wkxcass reviewed (commented):
I found no big issues regarding naming! 👍
Just a couple of nits to fix (including some minor lapses in coding standard adherence).

minhao23 commented on src/main/java/ollie/Parser.java:
```diff
@@ -0,0 +1,150 @@
+package ollie;
+
+import ollie.command.AddCommand;
+import ollie.command.Command;
```
good job making each command a class!

minhao23 commented on src/main/java/ollie/Ui.java:
```diff
@@ -0,0 +1,95 @@
+        System.out.println("____________________________________________________________\n"
+                + s
+                + "\n____________________________________________________________");
+    }
```
small issue, but there are places where singular nouns are used instead of plural ones for list

minhao23 commented on src/main/java/ollie/exception/OllieException.java:
```diff
@@ -0,0 +1,10 @@
+package ollie.exception;
```
good job placing all your exceptions in a package! applies to the other parts of your code as well

minhao23 commented on src/main/java/ollie/task/Todo.java:
```diff
@@ -0,0 +1,25 @@
+     *
+     * @return Formatted String.
+     */
+    public String getFormattedString() {
```
great use of polymorphism!

minhao23 reviewed (commented):
Great use of OOP principles, LGTM, small issues with tense but very well done!

NgZiXin committed:
Add GUI

NgZiXin committed:
Merge branch 'branch-Level-10'

NgZiXin committed:
Fix styling errors

NgZiXin committed:
Add basic assertions

Some assumptions are made in development, such as attribute tasks in TaskList being not null when either TaskList::IsIndexValid or TaskList::toString is called.

Using java assertions at appropriate palces ensures these assumptions are true when they are being.

NgZiXin committed:
Fix styling errors

NgZiXin committed:
Improve code quality

Code is not structured in a readable fashion and principles such as SLAP is being violated.

Improve code quality by resolving these issues help the readers to understand code better.

NgZiXin committed:
Merge pull request #1 from NgZiXin/branch-A-CodeQuality

Improve code quality.

NgZiXin committed:
Merge with main branch

NgZiXin committed:
Merge pull request #2 from NgZiXin/branch-A-Assertions

Merge Assertion branch

NgZiXin committed:
Add yml for continuous integration

NgZiXin committed:
Add undo feature

NgZiXin committed:
Merge branch 'BCD-extension'

NgZiXin committed:
Add additional error handling and improve code qualiy

Add more error handling fucntionalities in parser class and ensure SLAP is followed.

NgZiXin committed:
Edit header comments

NgZiXin committed:
Add user guide

NgZiXin committed:
Edit header comments

NgZiXin committed:
Edit header comments

NgZiXin committed:
Edit ReadMe

There are a lot of unnecessary spaces and mark down within the
ReadMe.

Removing all these formatting issues ensure that it can have
a clean look to it.

NgZiXin committed:
Create Jar File

There was an issue with the relative path where the filepath could
not be found and a new file could not be created.

Resolving the issue within the storage class allowed me to use
gradle to build a jar file that can be executed at any folder.

NgZiXin committed:
Reoragnise README.md

NgZiXin committed:
Reformat README.md

NgZiXin committed:
Reformat README.md

NgZiXin committed:
Edit gradle.yml

NgZiXin committed:
Edit README.md

NgZiXin committed:
Add exception handling for empty list

NgZiXin committed:
Update Jar file

NgZiXin committed:
Edit ReADME.md

NgZiXin committed:
Edit ReADME.md

NgZiXin committed:
Edit ReADME.md

NgZiXin committed:
Edit README.md

NgZiXin committed:
Fix checksytle errors
